src/Raven.Client/Documents/Session/ClusterTransactionSession.cs

```diff
@@ -321,16 +322,31 @@ internal CompareExchangeValue<T> GetValue<T>(DocumentConventions conventions)
                                 throw new InvalidOperationException("Value cannot be null.");
 
                             T entity = default;
-                            if (_originalValue != null && _originalValue.Value != null)
+                            IMetadataDictionary metadata = null;
+                            if (_originalValue != null)
                             {
-                                var type = typeof(T);
-                                if (type.IsPrimitive || type == typeof(string))
-                                    _originalValue.Value.TryGet(Constants.CompareExchange.ObjectFieldName, out entity);
-                                else
-                                    entity = conventions.Serialization.DefaultConverter.FromBlittable<T>(_originalValue.Value, _key);
+                                if (_originalValue.Value != null)
+                                {
+                                    var type = typeof(T);
+                                    if (type.IsPrimitive || type == typeof(string))
+                                        _originalValue.Value.TryGet(Constants.CompareExchange.ObjectFieldName, out entity);
+                                    else
+                                        entity = conventions.Serialization.DefaultConverter.FromBlittable<T>(_originalValue.Value, _key);
+                                }
+
+                                if (_originalValue.Metadata != null)
+                                {
+                                    metadata = new MetadataAsDictionary();
+                                    foreach (var kvp in _originalValue.Metadata)
+                                    {
+                                        var key = kvp.Key;
+                                        var val = kvp.Value;
+                                        metadata[key] = val;
+                                    }
+                                }
```

Comment:
why do we need that? and not passing `_originalValue?.Metadata` directly?

Reply:
Because you want to add it the modification, not modify it

Reply:
It works also with the ```_originalValue?.Metadata``` instead, but I don't know if it's a right attitude.